chain/chain/src/chain.rs

```diff
@@ -1312,6 +1314,57 @@ impl Chain {
         Ok(state_part)
     }
 
+    pub fn get_state_response_by_request(
+        &mut self,
+        shard_id: ShardId,
+        sync_hash: CryptoHash,
+        need_header: bool,
+        parts: StateRequestParts,
+        // TODO discuss: can 'static flood the memory?
+    ) -> BoxFuture<'static, Result<ShardStateSyncResponse, Error>> {
```

Comment:
I don't think use `'static` is a good idea. Why not add a lifetime parameter?

Reply:
I don't think it is a good idea, this pattern is used in other places too, should they be revisited?

Reply:
Where else is it used?

Reply:
in `trait NetworkAdapter` and `trait AsyncUser`